### PR TITLE
[NVIDIA] CMAKE_CUDA_ARCHITECTURES support and new architectures

### PR DESCRIPTION
**Pull request by apavliuk-altran**
### Details:
- *added `CMAKE_CUDA_ARCHITECTURES` support and new architectures*
### Ticket:
- 102638

### CODE REVIEW

File: modules/nvidia_plugin/src/CMakeLists.txt [hunk 1]
```diff
@@ -37,11 +37,7 @@ ie_add_plugin(NAME ${TARGET_NAME}
 # Enable support of CC for the plugin
 ie_mark_target_as_cc(${TARGET_NAME})
 
-if (${CUDAToolkit_VERSION_MAJOR} LESS_EQUAL "10")
-    set_property(TARGET ${OBJ_NAME} PROPERTY CUDA_ARCHITECTURES 30 35 50 60 72)
-else()
-    set_property(TARGET ${OBJ_NAME} PROPERTY CUDA_ARCHITECTURES 35 50 60 72)
-endif()
+set_property(TARGET ${OBJ_NAME} PROPERTY CUDA_ARCHITECTURES ${CMAKE_CUDA_ARCHITECTURES})
```
Comment (nkogteva) [inline modules/nvidia_plugin/src/CMakeLists.txt:40]: Can we add printing of `CMAKE_CUDA_ARCHITECTURES` just to make it visible in cmake logs?

File: modules/nvidia_plugin/CMakeLists.txt [hunk 2]
```diff
@@ -91,6 +97,16 @@ add_library(CUDA::cutensor SHARED IMPORTED)
 set_target_properties(CUDA::cutensor PROPERTIES IMPORTED_LOCATION "${CUTENSOR_PATH}")
 set_target_properties(CUDA::cutensor PROPERTIES IMPORTED_IMPLIB "${CUTENSOR_PATH}")
 
+if(NOT DEFINED CMAKE_CUDA_ARCHITECTURES)
+    if(${CUDAToolkit_VERSION_MAJOR} LESS_EQUAL 10)
+        set(CMAKE_CUDA_ARCHITECTURES 30 35 50 60 72)
+    elseif(${CUDAToolkit_VERSION_MAJOR} EQUAL 11 AND ${CUDAToolkit_VERSION_MINOR} LESS_EQUAL 7)
+        set(CMAKE_CUDA_ARCHITECTURES 35 50 60-virtual 61 72-virtual 75 86)
+    else()
+        set(CMAKE_CUDA_ARCHITECTURES 35 50 60-virtual 61 72-virtual 75 86 89-virtual 90-virtual)
```
[the remark this answers is not in the document]
Comment (apavliuk-altran) [reply modules/nvidia_plugin/CMakeLists.txt]: Removed 3.5 architecture for CUDA 12+

File: modules/nvidia_plugin/src/CMakeLists.txt (hunk 1 above)
Comment (apavliuk-altran) [reply modules/nvidia_plugin/src/CMakeLists.txt:40]: Done (in modules/nvidia_plugin/CMakeLists.txt)

**Comment by apavliuk-altran** [comment]
@nkogteva @vladimir-paramuzov
With this configuration the size of `libopenvino_nvidia_gpu_plugin.so` increased from 58 to 79 MB (for CUDA 11.8), which is expected as we are added more architectures.
Is such size acceptable?

**Comment by nkogteva** [comment]
> @nkogteva @vladimir-paramuzov With this configuration the size of `libopenvino_nvidia_gpu_plugin.so` increased from 58 to 79 MB (for CUDA 11.8), which is expected as we are added more architectures. Is such size acceptable?

As we discussed on the meeting we have to add opportunity for the user to override compute capabilities. I suppose it will help to reduce binary size. Also maybe there is a sense to update documentation with section about compute capabilities (for example, how to check compute capabilitues of the GPU).  It make sense?

**Comment by nkogteva** [comment]
Also please rebase to fix tests https://github.com/openvinotoolkit/openvino_contrib/pull/594

**Comment by apavliuk-altran** [comment]
> > @nkogteva @vladimir-paramuzov With this configuration the size of `libopenvino_nvidia_gpu_plugin.so` increased from 58 to 79 MB (for CUDA 11.8), which is expected as we are added more architectures. Is such size acceptable?
> 
> As we discussed on the meeting we have to add opportunity for the user to override compute capabilities. I suppose it will help to reduce binary size. Also maybe there is a sense to update documentation with section about compute capabilities (for example, how to check compute capabilitues of the GPU). It make sense?

**Comment by apavliuk-altran** [comment]
> > @nkogteva @vladimir-paramuzov With this configuration the size of `libopenvino_nvidia_gpu_plugin.so` increased from 58 to 79 MB (for CUDA 11.8), which is expected as we are added more architectures. Is such size acceptable?
> 
> As we discussed on the meeting we have to add opportunity for the user to override compute capabilities. I suppose it will help to reduce binary size. Also maybe there is a sense to update documentation with section about compute capabilities (for example, how to check compute capabilitues of the GPU). It make sense?

We updated the documentation.

**Review by nkogteva — APPROVED** [review]

**Pull request merged**